pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/codestyle/UnnecessaryFullyQualifiedNameRule.java

```diff
@@ -170,11 +178,31 @@ private boolean isAvoidingConflict(final JavaNode node, final String name,
             }
         }
 
-        // Is it a conflict with a class in the same file?
         final String unqualifiedName = name.substring(name.lastIndexOf('.') + 1);
         final int unqualifiedNameLength = unqualifiedName.length();
+
+        // There could be a conflict between an import on demand and another import, e.g.
+        // import One.*;
+        // import Two.Problem;
+        // Where One.Problem is a legitimate qualification
+        if (firstMatch.isImportOnDemand() && !firstMatch.isStatic()) {
+            for (ASTImportDeclaration importDeclaration : imports) {
+                if (!Objects.equals(importDeclaration, firstMatch)
```

Comment:
I think, using Object.equals here is misleading - we are actually comparing the references (ASTImportDeclaration doesn't have an own equals method implementation).
So, I'd use simply `importDeclaration != firstMatch`

Reply:
Yes, it would be clearer that way. We'll have to suppress CompareObjectsWithEquals though